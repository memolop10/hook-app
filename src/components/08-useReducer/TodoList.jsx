import React from 'react'
import TodoListItem from './TodoListItem'

const TodoList = ({ todos, handleDelete, handleToggle }) => {
    return (
            <ul className="list-group list-group-flush">
                {
                    todos.map( (todo, index) => (
                        <TodoListItem 
                            todo={ todo }
                            index={ index }
                            key={ todo.id }
                            handleDelete= { handleDelete }
                            handleToggle= { handleToggle }
                        />

                    ))
                }
            </ul>
    )
}

export default TodoList
