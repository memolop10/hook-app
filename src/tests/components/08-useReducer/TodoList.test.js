import React from 'react'
import { shallow } from 'enzyme'
import TodoList from '../../../components/08-useReducer/TodoList'
import { demoTodos } from '../fixtures/demoTodos'
// import TodoListItem from '../../../components/08-useReducer/TodoListItem'


describe('Test in component <TodoList />', () => {
    const handleDelete = jest.fn()
    const handleToggle = jest.fn()

    const wrapper = shallow( 
        <TodoList 
            todos={ demoTodos }
            handleDelete= { handleDelete }
            handleToggle= { handleToggle }
        /> 
    )
    
    test('should render correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('should have two <TodoList />', () => {

        //  console.log(wrapper.find('TodoListItem').at(1).props())  
         expect( wrapper.find('TodoListItem').length ).toBe( demoTodos.length )
         expect( wrapper.find('TodoListItem').at(1).prop('handleDelete') ).toEqual( expect.any( Function ) )
    })
})
