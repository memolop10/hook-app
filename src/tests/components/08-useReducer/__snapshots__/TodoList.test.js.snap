// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <TodoList /> should render correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="11"
    todo={
      Object {
        "done": false,
        "id": 11,
        "todo": "Comprar leche",
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="12"
    todo={
      Object {
        "done": false,
        "id": 12,
        "todo": "Aprender de la vida",
      }
    }
  />
</ul>
`;
